pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -6571,30 +6597,45 @@ spec:
 	prt := newPipelineRunTest(t, d)
 	defer prt.Cancel()
 
-	run1, _ := prt.reconcileRun("foo", "pipelinerun-param-invalid-result-variable", nil, true)
-	run2, _ := prt.reconcileRun("foo", "pipelinerun-pipeline-result-invalid-result-variable", nil, true)
-	run3, _ := prt.reconcileRun("foo", "pipelinerun-with-optional-workspace-validation", nil, true)
-
-	cond1 := run1.Status.GetCondition(apis.ConditionSucceeded)
-	cond2 := run2.Status.GetCondition(apis.ConditionSucceeded)
-	cond3 := run3.Status.GetCondition(apis.ConditionSucceeded)
-
-	for _, c := range []*apis.Condition{cond1, cond2, cond3} {
-		if c.Status != corev1.ConditionFalse {
-			t.Errorf("expected Succeeded/False condition but saw: %v", c)
-		}
-	}
-
-	if cond1.Reason != ReasonInvalidTaskResultReference {
-		t.Errorf("expected invalid task reference condition but saw: %v", cond1)
-	}
-
-	if cond2.Reason != ReasonInvalidTaskResultReference {
-		t.Errorf("expected invalid task reference condition but saw: %v", cond2)
+	testCases := []struct {
+		name      string
+		wantErr   string
+		condition apis.ConditionType
+		reason    string
+	}{
+		{
+			name:      "pipelinerun-param-invalid-result-variable",
+			wantErr:   ("expected invalid task reference condition but saw: %v"),
+			condition: apis.ConditionSucceeded,
+			reason:    ReasonInvalidTaskResultReference,
+		}, {
+			name:      "pipelinerun-pipeline-result-invalid-result-variable",
+			wantErr:   ("expected invalid task reference condition but saw: %v"),
+			condition: apis.ConditionSucceeded,
+			reason:    ReasonInvalidTaskResultReference,
+		}, {
+			name:      "pipelinerun-with-optional-workspace-validation",
+			wantErr:   ("expected optional workspace not supported condition but saw: %v"),
+			condition: apis.ConditionSucceeded,
+			reason:    ReasonRequiredWorkspaceMarkedOptional,
+		}, {
+			name:      "pipelinerun-matrix-param-invalid-type",
+			wantErr:   ("expected invalid matrix parameter types condition but saw: %v"),
+			condition: apis.ConditionSucceeded,
+			reason:    ReasonInvalidMatrixParameterTypes,
+		},
 	}
-
-	if cond3.Reason != ReasonRequiredWorkspaceMarkedOptional {
-		t.Errorf("expected optional workspace not supported condition but saw: %v", cond3)
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			run, _ := prt.reconcileRun("foo", tc.name, nil, true)
+			c := run.Status.GetCondition(tc.condition)
+			if c.Status != corev1.ConditionFalse {
+				t.Errorf("expected Succeeded/False condition but saw: %v", c)
+			}
+			if c.Reason != tc.reason {
+				t.Errorf(tc.wantErr, c.Reason)
```

Comment:
tc.wantErr is pretty confusing here, I would remove tc.wantErr and just do
`t.Errorf("want reason %s but got %s)`